src/include/compat.h

```diff
@@ -112,4 +112,33 @@
 #define MSG_DONTWAIT MSG_NONBLOCK
 #endif
 
+#if defined(HAVE_PTHREAD_SETNAME_NP)
+  #if defined(__APPLE__)
+    #define pthread_setname_np(thread, name) ({ \
+      int __result = 0;                         \
+      if (thread == pthread_self())             \
+        __result = pthread_setname_np(name)     \
+      __result; })
+  #else
+    #define pthread_setname_np(thread, name) pthread_setname_np(thread, name)
```

Comment:
On 1-7-2016 17:18, Kefu Chai wrote:
> In src/include/compat.h
> <https://github.com/ceph/ceph/pull/9763#discussion_r69315093>:
> 
>> @@ -112,4 +112,33 @@
>>  #define MSG_DONTWAIT MSG_NONBLOCK
>>  #endif
>>  
>> +#if defined(HAVE_PTHREAD_SETNAME_NP)
>> +  #if defined(__APPLE__)
>> +    #define pthread_setname_np(thread, name) ({ \
>> +      int __result = 0;                         \
>> +      if (thread == pthread_self())             \
>> +        __result = pthread_setname_np(name)     \
>> +      __result; })
>> +  #else
>> +    #define pthread_setname_np(thread, name) pthread_setname_np(thread, name)
> 
> we don't need this |#define| in this case.

Not only is it not needed, but I'd expect the preprocessor to start looping.
But I this that the first name needs a _ between set_name.